### PR TITLE
update several versions (upp, metplus)

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
add upp/8.2.0
met/10.1.2
metplus/4.1.4

### CODE REVIEW

File: var/spack/repos/builtin/packages/upp/package.py [hunk 1]
```diff
@@ -15,10 +15,11 @@ class Upp(CMakePackage):
 
     homepage = "https://github.com/NOAA-EMC/UPP"
     git = "https://github.com/NOAA-EMC/UPP.git"
-    url = "https://github.com/NOAA-EMC/UPP/archive/refs/tags/upp_v10.0.10.tar.gz"
+    url = "https://github.com/NOAA-EMC/UPP/archive/refs/tags/upp_v8.2.0.tar.gz"
 
     maintainers = ["kgerheiser", "edwardhartnett", "Hang-Lei-NOAA"]
 
+    version("8.2.0", sha256="38de2178dc79420f42aa3fb8b85796fc49d43d66f90e5276e47ab50c282627ac")
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/upp/package.py]: Versions are ordered newest to oldest, please move this to below 10.0.9

File: var/spack/repos/builtin/packages/upp/package.py [hunk 2]
```diff
@@ -15,10 +15,11 @@ class Upp(CMakePackage):
 
     homepage = "https://github.com/NOAA-EMC/UPP"
     git = "https://github.com/NOAA-EMC/UPP.git"
-    url = "https://github.com/NOAA-EMC/UPP/archive/refs/tags/upp_v10.0.10.tar.gz"
+    url = "https://github.com/NOAA-EMC/UPP/archive/refs/tags/upp_v8.2.0.tar.gz"
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/upp/package.py]: No need to change.

**Review by climbfuji — CHANGES_REQUESTED** [review]

**Comment by climbfuji** [comment]
Style tests are failing, can you copy & paste the black format changes into a patch file and apply it? Or let me know if you need help. Thanks!

https://github.com/NOAA-EMC/spack/actions/runs/3175485649/jobs/5173580515

File: var/spack/repos/builtin/packages/upp/package.py (hunk 1 above)
Comment (Hang-Lei-NOAA) [reply var/spack/repos/builtin/packages/upp/package.py]: solved

File: var/spack/repos/builtin/packages/upp/package.py (hunk 2 above)
Comment (Hang-Lei-NOAA) [reply var/spack/repos/builtin/packages/upp/package.py]: solved it and the black format

**Comment by Hang-Lei-NOAA** [comment]
@climbfuji  need your help on this: 
I try two ways for these changes, only changes in upp is fine.
 (1) my original edit is using spack edit in runing checkout, and copied to the push checkout.
(2) I followed your suggestion to clean the black format by doing:
% diff -u original_package.py modifed_package.py >package.patch
% patch original_package.py package.patch
Then commit the package.py.

Both ways failed for met and metplus.

**Comment by climbfuji** [comment]
@Hang-Lei-NOAA I created a PR to update your branch and fix the remaining style errors:

https://github.com/Hang-Lei-NOAA/spack/pull/1

**Comment by Hang-Lei-NOAA** [comment]
Hi, Dom, Please do so. Thanks

**Review by climbfuji — APPROVED** [review]

**Comment by climbfuji** [comment]
Note that this PR only updates upp and metplus, not met (the description says upp, met, metplus).

**Pull request merged**